public/js/post/post.js

```diff
@@ -63,47 +63,57 @@ var MultipleCourseCode = React.createClass({
             infoOpened: false
         }
     },
+
+    componentDidMount: function() {
+        this.setState({completedTextBoxes: this.state.completedTextBoxes + this.props.data.courses.length},
+            this.checkIfCompleted);
+    },
     
     toggleFullInfo: function() {
         this.setState({infoOpened: !this.state.infoOpened});
     },
 
     checkIfCompleted: function() {
-       this.setState({completed: this.state.completedTextBoxes === this.props.data.textBoxNumber});
+        this.setState({completed: this.state.completedTextBoxes === this.props.data.textBoxNumber});
     },
 
     handleKeyDown: function(e) {
         if (e.keyCode === 13) {
-            if (this.state.completedTextBoxes < this.props.data.textBoxNumber) {
-                 this.setState({completedTextBoxes: this.state.completedTextBoxes +=1})
+            if (this.state.completedTextBoxes <= this.props.data.textBoxNumber + 1) {
+                if (e.target.defaultValue === '' && e.target.value !== '') {
+                    this.setState({completedTextBoxes: this.state.completedTextBoxes += 1});
```

Comment:
Oops, different part of the code than the last time, but still same error. Thanks!